Issue 4460 - python3-lib389 ignore the configuration parameters from …

aadhikar opened the pull request:
…/etc/openldap/ldap.conf

Description: python3-lib389 v1.4.3 seems to ignore the configuration parameters from /etc/openldap/ldap.conf.

Relates: https://github.com/389ds/389-ds-base/issues/4460

Reviewed by: ???

aadhikar committed:
Issue 4460 - python3-lib389 ignore the configuration parameters from /etc/openldap/ldap.conf

Description: python3-lib389 v1.4.3 seems to ignore the configuration parameters from /etc/openldap/ldap.conf.

Relates: https://github.com/389ds/389-ds-base/issues/4460

Reviewed by: ???

aadhikar commented on dirsrvtests/tests/suites/clu/dsconf_test.py:
```diff
@@ -0,0 +1,87 @@
+            fd.write(line)
+        fd.close()
+
+    def fin():
```
Not sure if this is the right way of handling this, need suggestions, Thanks!

Firstyear replied:
yeah, this isn't a good way to do it. Remember, the test could be run as someone who is not-root (ie dirsrv user), and each time this test is run it's going to keep adding this. Imagine the test fails and fin isn't run, then it will remain for example? 

A better way is that the libldap tools read from the environment with each variable in ldap.conf prefixed with LDAP. 

For example  `LDAPTLS_CACERT=/path/to/cert dsconf ....`

that should allow an alternate way to set this value that is non destructive and should have the same effect. 

Hope that helps!

aadhikar replied:
@Firstyear Hey thanks for your review, this bug is related to ```ldap.conf``` specifically so I think it's better to use ```LDAPCONF``` in this way we can be sure that settings from ```ldap.conf``` are respected correctly, let me know your thoughts on this. :)

aadhikar committed:
Added LDAPCONF env variable to be used instead of writing in ldap.conf.
Added HOME env variable to be used as path (to make it work with non-root users).

sgouvern reviewed (approved):
Ack

sgouvern reviewed (approved):
Ack

sgouvern reviewed (commented):
Ack

sgouvern reviewed (commented):
Ack
